python/engine/evaluation.py

```
@@ -13,14 +13,182 @@
+
+    gen_text = tokenizer.decode(tokens[len(prompt_ids):]) if tokenizer else ""
+
+    # Compute summary stats
+    n = len(losses)
+    if n >= 10:
+        first10 = sum(losses[:10]) / 10
+        last10 = sum(losses[-10:]) / 10
+        # Simple linear regression for slope
+        mean_x = (n - 1) / 2.0
+        mean_y = sum(losses) / n
+        num = sum((i - mean_x) * (l - mean_y) for i, l in enumerate(losses))
```

Ruff (0.15.2)

[error] Ambiguous variable name: `l`

(E741)